scripts/globals/bcnm.lua

```diff
@@ -598,6 +599,9 @@ function checkReqs(player, npc, bfid, registrant)
     {
         [ 897] = function() return ( player:hasKeyItem(tpz.ki.WHISPER_OF_THE_WYRMKING)                                                      ) end, -- Quest: The Wyrmking Descends
         [ 928] = function() return ( player:hasCompletedMission(COP, mi.cop.ANCIENT_VOWS) or (cop == mi.cop.ANCIENT_VOWS and copStat >= 2)  ) end, -- Quest: Ouryu Cometh
+        [1057] = function() return ( player:hasCompletedQuest(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) or 
+		                             player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and 
+									 player:getCharVar('ApocalypseNigh') == 4                                                               ) end, -- ApocalypseNigh
```

Comment:
You actually might need parens here to clarify the order of operations; since
they can enter `if`:
they've completed `or` **(** are on quest `and` on step 4 **)**
```lua
    player:hasCompletedQuest(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) or 
    (player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and 
    player:getCharVar('ApocalypseNigh') == 4)
```

Reply:
It's okay as coded.  AND happens before OR.